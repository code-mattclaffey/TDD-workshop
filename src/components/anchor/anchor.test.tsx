import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Anchor } from '.';

afterEach(() => {
    cleanup();
});

describe('<Anchor />', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <Anchor url="#">a standard link</Anchor>,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match target blank snapshot', () => {
        const { container } = render(
            <Anchor url="#" target="_blank">
                a standard link
            </Anchor>,
        );

        expect(container).toMatchSnapshot();
    });
});
