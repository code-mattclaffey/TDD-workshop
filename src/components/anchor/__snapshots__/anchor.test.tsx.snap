// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> should match snapshot 1`] = `
<div>
  <a
    class="e-link"
    href="#"
  >
    a standard link
  </a>
</div>
`;

exports[`<Anchor /> should match target blank snapshot 1`] = `
<div>
  <a
    class="e-link"
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    a standard link
  </a>
</div>
`;
